packages/business-rules/src/rules/appeal/deadline-date.test.js

```diff
@@ -1,19 +1,53 @@
-const { addWeeks, endOfDay } = require('date-fns');
+const { addWeeks, endOfDay, addMonths } = require('date-fns');
 const deadlineDate = require('./deadline-date');
-const { APPEAL_ID } = require('../../constants');
+const { APPEAL_ID, APPLICATION_DECISION } = require('../../constants');
 
 describe('business-rules/appeal/deadline-date', () => {
   it('should throw an error if an invalid date is given', () => {
-    expect(() => deadlineDate(null)).toThrow('The given date must be a valid Date instance');
+    expect(() => deadlineDate(null, APPEAL_ID.HOUSEHOLDER, APPLICATION_DECISION.REFUSED)).toThrow(
+      'The given date must be a valid Date instance',
+    );
+  });
+
+  it('should throw an error if an invalid appeal type is given', () => {
+    expect(() => deadlineDate(new Date(), '100', APPLICATION_DECISION.REFUSED)).toThrow(
+      '100 is not a valid appeal type',
+    );
   });
 
   it('should throw an error if an invalid appeal type is given', () => {
-    expect(() => deadlineDate(new Date(), '100')).toThrow('100 is not a valid appeal type');
+    expect(() => deadlineDate(new Date(), APPEAL_ID.HOUSEHOLDER, '100')).toThrow(
+      '100 must be a valid application decision',
+    );
+  });
+
+  it('should return the correct date for an appeal type', () => {
+    expect(deadlineDate(new Date(), APPEAL_ID.HOUSEHOLDER, APPLICATION_DECISION.REFUSED)).toEqual(
+      addWeeks(endOfDay(new Date()), 12),
+    );
   });
 
   it('should return the correct date for an appeal type', () => {
-    expect(deadlineDate(new Date(), APPEAL_ID.HOUSEHOLDER)).toEqual(
+    expect(deadlineDate(new Date(), undefined, APPLICATION_DECISION.REFUSED)).toEqual(
       addWeeks(endOfDay(new Date()), 12),
     );
   });
+
+  it('should return the correct date for an appeal type', () => {
```

Comment:
It would make his test clearer if the description included that this test was checking the scenario of no appeal type or application decision being given

Reply:
Updated